src/Core/Services/RestService.cs

```diff
@@ -372,6 +372,13 @@ public string GetRouteAfterPathBase(string route)
             // The RuntimeConfigProvider enforces the expectation that the configured REST path starts with a
             // forward slash '/'.
             configuredRestPathBase = configuredRestPathBase.Substring(1);
+
+            // Consider a typical URL of the form: <scheme>://<domain-name>/<path_to_the_resource>.
+            // The variable route contains the value of <path-to-resource> section.   
+            // Since, <scheme>://<domain-name>//<path_to_the_resource> is a valid URL, leading slashses are removed before comparing it with
+            // incoming request's path.
+            route = route.TrimStart('/');
```

Comment:
if `//api/<path_to_the_resource>` is a valid url, why are we modifying the user entered URL here? An invalid URL should route to HTTP 404 Not Found. By trimming this URL we are assuming (perhaps incorrectly) what the user intended url should be.

Consider the double slash with different placement in the URL: `https://localhost:5001/api//Book`. That is an invalid route for accessing book, because the name of the entity is blank `//`, and Book would be interpreted as the primary key. This example is handled by preexisting code in the return statement at the end of this method `return route.Substring(configuredRestPathBase.Length).TrimStart('/');` I think this should probably be removed too and instead result in 404.

Reply:
TrimStart in the return statement was intended to only remove a slash if only 1 existed, not >1.

Reply:
Removed logics related to trimming leading slashes. Since this discussion is tangential to the intention of this PR, opened [this](https://github.com/Azure/data-api-builder/issues/1640) for tracking the same. Based on the conclusion there, a follow-up PR can take care of those changes.